core/src/main/java/com/graphhopper/storage/CHProfile.java

```diff
@@ -20,14 +20,18 @@
 public class CHProfile {
     private final Weighting weighting;
     private final boolean edgeBased;
-    private final int uTurnCosts;
 
     public static CHProfile nodeBased(Weighting weighting) {
-        return new CHProfile(weighting, TraversalMode.NODE_BASED, INFINITE_U_TURN_COSTS);
+        // todonow: should we do such a check (check if the weighting has turn costs ?)
```

Comment:
Any opinion here? Its no longer easy to do this check, we would need something like `weighting.hasTurnCosts()` (and maybe check if `TurnCostProvider` is a `NoTurnCostProvider`. But that doesn't seem very clean. Also using a `TurnCostProvider` and then choosing `nodeBased` is not necessarily an error (but might be useful to receive a warning...).

Reply:
I removed the check, it would be nice to have, but not really important either imo.